examples: added instructor + burr example; youtube transcription (#313)
DAGWorks-Inc/burr: examples/youtube-to-social-media-post/README.md

# Build trustworthy LLM agents & applications for production with Instructor + Burr

The challenge with large language models (LLMs) is handling the 5% of the time they say crazy things. Being able to debug why an output is bad and having tools for fixing are critical requirements for making LLM features / agents trustworthy and available to users.

> In this example, you'll learn how `instructor` can make LLM reliability produce structured outputs, and `burr` helps you introspect and debug your application.

Reviewer: ```suggestion
In this example, you'll learn how `instructor` can make LLM reliability produce structured outputs, and `burr` helps you introspect, debug, and create test cases for your application.
```

> In this example, you'll learn how `instructor` can make LLM reliability produce structured outputs, and `burr` helps you introspect, debug, and create tests for your application.

Reviewer: ```suggestion
In this example, you'll learn how `instructor` can make LLMs reliability produce structured outputs, and `burr` helps you introspect, debug, and create tests for your application.
```

> In a few lines of code, you can query an LLM API and can create powerful productivity utilities. However, user-facing features deserve much more scrutiny, which requires tooling and solving complex engineering problems.

Reviewer: ```suggestion
While all it takes to create a cute LLM/agent demo is a few lines of code, production user-facing features aren't as simple. To build trust that they will always work as intended, they need much more scrutiny which requires instrumentation along with a fast development loop; you're not going to get it 100% right the first time, so you need to be able to iterate to get there quickly so you can ship that feature with confidence.
```

> While a LLM/agent demo takes a few lines of code, production user-facing features aren't as simple. To build trust, they need to work as intended consistently, requiring much more scrutiny and the right instrumentation. You're agent won't be perfect from the start, and tools should help you improve it iteratively and ship updates with confidence.

Reviewer: ```suggestion
While a LLM/agent demo takes a few lines of code, production user-facing features aren't as simple. To build trust, they need to work as intended consistently, requiring much more scrutiny and the right instrumentation. Your agent won't be perfect from the start, and tools should help you improve it iteratively and ship updates with confidence.
```

Building our app with Burr provides several benefits that we'll detail next:

> - **Observability**: monitor in real-time and log the execution of your `Application` and view it in Burr's web user interface.

Reviewer: ```suggestion
- **Observability**: monitor in real-time and log the execution of your `Application` and view it in Burr's web user interface, and also use that data to easily create test cases to build guardrails. 
```

Reviewer: or put the note under persistence... just want to match what's in the notebook.

> - **Persistence**: At any point, you can save the application `State`. This allows to create user sessions (e.g., the conversation history menu in ChatGPT), which helps developers investigate bugs and test potential solutions.

Reviewer: ```suggestion
- **Persistence**: At any point, you can save the application `State`. This allows to create user sessions (e.g., the conversation history menu in ChatGPT), which helps developers investigate bugs, iterate on different code paths, & test potential solutions in an integrated manner.
```

> - **Persistence**: At any point, you can save the application `State`. This allows to create user sessions (e.g., the conversation history menu in ChatGPT), which helps developers investigate bugs, iterate over possible code paths, and test potential solutions in an integrated manner.

Reviewer: ```suggestion
- **Persistence**: At any point, you can save the application `State`. This allows you to create user sessions (e.g., the conversation history menu in ChatGPT), which helps developers investigate bugs, iterate over possible code paths, and test potential solutions in an integrated manner.
```
